### DISCUSSION TITLE
Feedback: There are no pricing information fetched from resources of some regions

### DISCUSSION THREAD

**Discussion by jolo-dev** [tailwarden/komiser#1155] [Ideas]
Most AWS resources like Lambda and DynamoDB lack a [Pricing API](https://www.aws-services.info/pricing.html), resulting in no pricing information for certain regions.
Actually, you need some sort of a web scraper to get the "real" information until the Pricing API is available in all region:
https://aws.amazon.com/lambda/pricing/

What's your thought on that?

**Comment by mlabouardy** [comment]
cc @Azanul @Traxmaxx

**Comment by Azanul** [comment]
@jolo-dev Both AWS Lambda & DynamoDB are supported by pricing API and are even used currently in Komiser

**Reply by jolo-dev** [reply]
@Azanul, Yes, there is a pricing API, but only for 4 regions (eu-central-1, us-east-1, cn-northwest-1, ap-south-1)
https://www.aws-services.info/pricing.html

That is why `eu-central-2` would result in zero.
At first glance, each continent has the same pricing (us-east-1 === us-east-2 === us-west-1, same for eu-central).

So the idea would be that here, it should take the pricing from the geographical continent instead of resulting to an error (same for different services):

https://github.com/tailwarden/komiser/blob/6cffe2945c718d3baebf1a9ca01fe2db7cecb3cb/providers/aws/lambda/functions.go#L40C2-L54C1

**Reply by Azanul** [reply]
@jolo-dev I don't think that's the case. The pricing API is available at these 4 regions but the data returned is the same which does cover all regions (in which the particular service is available). While fetching using pricing client(which can be initialized with any of the 4 regions), we can filter the results by specifying the `regionCode` field (this is where we mention the region for which we want the prices). AFAIK the results by default ,without `regionCode` field, are not tied to the region which is used to initialized the pricing client.

I've attached the pricing output of AWSLambda, please check it out and let me know if you still have a confusion or you've found something contrary to what I described.
[AWSLambda-pricingOutput.json](https://github.com/tailwarden/komiser/files/13239092/AWSLambda-pricingOutput.json)
